src/helpers.js

```diff
@@ -1,46 +1,83 @@
 const p = require('path')
-const babel = require('babel-core')
-const template = require('babel-template')
-// const printAST = require('ast-pretty-print')
+const fs = require('fs')
 const requireFromString = require('require-from-string')
 
 module.exports = {
-  replace,
-  looksLike,
-  isCodegenComment,
+  transformAndRequire,
   getReplacement,
   stringToAST,
-  resolveModuleToString,
+  replace,
+  resolveModuleContents,
+  isCodegenComment,
   isPropertyCall,
+  looksLike,
 }
 
-function getReplacement({string: stringToCodegen, filename}) {
-  const {code: transformed} = babel.transform(stringToCodegen, {
+function transformAndRequire({code, filename}, babel) {
+  /**
+   * Transform passed in code to generated code
+   */
+  const {code: transformedCode} = babel.transform(code, {
     filename,
   })
-  const val = requireFromString(transformed, filename)
-  return stringToAST(val)
+
+  /**
+   * Execute the transformed code, as if it were required
+   */
+  const module = requireFromString(transformedCode, filename)
+
+  /**
+   * Allow for es modules (default export)
+   */
+  return module && module.__esModule ? module.default : module
+}
+
+function getReplacement({code, filename, args = []}, babel) {
+  /**
+   * Execute the code string to get the exported module
+   */
+  let module = transformAndRequire({code, filename}, babel)
+
+  /**
+   * If a function is epxorted, call it with args
+   */
+  if (typeof module === 'function') {
+    module = module(...args)
+  } else if (args.length) {
+    throw new Error(
+      `codegen module (${p.relative(
+        process.cwd(),
+        filename,
+      )}) cannot accept arguments because it does not export a function. You passed the arguments: ${args.join(
+        ', ',
+      )}`,
+    )
+  }
+
+  /**
+   * Convert whatever we got now (hopefully a string) into AST form
+   */
+  return stringToAST(module, babel)
 }
 
-function stringToAST(string) {
+function stringToAST(string, babel) {
   if (typeof string !== 'string') {
     throw new Error('codegen: Must module.exports a string.')
   }
-  return template(string, {
+  return babel.template(string, {
```

Comment:
If `state` was also passed through, `state.file.opts.parserOpts` could be accessed and something like the following could be done so the plugin list (and any other options) would inherit from the main babel config:

```js
return babel.template(string, { 
  sourceType:"module", 
  ...parserOpts, 
  preserveComments: true, 
  placeholderPattern: false 
})()
```